core/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -344,6 +346,45 @@ public int compare(Map.Entry<String, String> o1, Map.Entry<String, String> o2) {
 
     }
 
+    /**
+     * Checks and returns the status of the existing index in this store.
+     *
+     * @param out where infoStream messages should go. See {@link CheckIndex#setInfoStream(PrintStream)}
+     */
+    public CheckIndex.Status checkIndex(PrintStream out) throws IOException {
+        // We don't need to lock the directory here as we are not changing the index files.
+        final Lock noDirectoryLock = new Lock() {
```

Comment:
I wonder if we should still acquire the lock. I don't think we should execute this while the IW is open?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/27768/commits/f231042ffd970b99fa0037ad3e5a6ce1c181674b